src/wifipaf/WiFiPAFLayer.cpp

```diff
@@ -384,84 +392,143 @@ WiFiPAFLayer::GetHighestSupportedProtocolVersion(const PAFTransportCapabilitiesR
     return retVersion;
 }
 
-void WiFiPAFLayer::AddPafSession(const NodeId nodeId, const uint16_t discriminator)
+#define INVALID_PAF_SESSION_ID 0xff
+void WiFiPAFLayer::InitialPafInfo()
 {
-    for (auto PafInfoElm : PafInfoVect)
+    for (uint8_t i = 0; i < WIFIPAF_LAYER_NUM_PAF_ENDPOINTS; i++)
     {
-        if (PafInfoElm.nodeId == nodeId)
-        {
-            assert(PafInfoElm.discriminator == discriminator);
-            // Already exist
-            return;
-        }
+        cleanPafInfo(mPafInfoVect[i]);
     }
-    PafInfoVect.push_back({ .id = UINT32_MAX, .peer_id = UINT32_MAX, .nodeId = nodeId, .discriminator = discriminator });
-    ChipLogProgress(WiFiPAF, "WiFiPAF: Add session with nodeId: %lu, disc: %x, total %lu sessions", nodeId, discriminator,
-                    PafInfoVect.size());
 }
 
-void WiFiPAFLayer::AddPafSession(uint32_t id)
+void WiFiPAFLayer::cleanPafInfo(WiFiPAFSession & SessionInfo)
 {
-    for (auto PafInfoElm : PafInfoVect)
-    {
-        if (PafInfoElm.id == id)
-        {
-            // Already exist
-            return;
-        }
-    }
-    PafInfoVect.push_back({ .id = id });
-    ChipLogProgress(WiFiPAF, "WiFiPAF: Add session with id: %u, total %lu sessions", id, PafInfoVect.size());
+    memset(&SessionInfo, 0, sizeof(WiFiPAFSession));
+    SessionInfo.id            = UINT32_MAX;
+    SessionInfo.peer_id       = UINT32_MAX;
+    SessionInfo.nodeId        = UINT32_MAX;
+    SessionInfo.discriminator = UINT16_MAX;
+    return;
 }
 
-void WiFiPAFLayer::RmPafSession(uint32_t id)
+CHIP_ERROR WiFiPAFLayer::AddPafSession(PafInfoAccess accType, WiFiPAFSession & SessionInfo)
 {
-    for (std::vector<WiFiPAFSession>::iterator it = PafInfoVect.begin(); it != PafInfoVect.end(); it++)
+    uint8_t i;
+    uint8_t eSlotId              = INVALID_PAF_SESSION_ID;
+    WiFiPAFSession * pPafSession = nullptr;
+
+    // Check if the session has existed
+    for (i = 0; i < WIFIPAF_LAYER_NUM_PAF_ENDPOINTS; i++)
     {
-        if (it->id == id)
+        pPafSession = &mPafInfoVect[i];
+        switch (accType)
         {
-            PafInfoVect.erase(it);
-            return;
-        }
+        case PafInfoAccess::kAccNodeInfo:
+            if (pPafSession->nodeId == SessionInfo.nodeId)
+            {
+                assert(pPafSession->discriminator == SessionInfo.discriminator);
```

Comment:
Direct asserts are forbidden. Please use `VerifyOrDie`

Reply:
I will change it